src/script/lua_api/l_inventory.cpp

```diff
@@ -241,6 +241,36 @@ int InvRef::l_set_list(lua_State *L)
 	return 0;
 }
 
+// add_list(self, listname, list)
+int InvRef::l_add_list(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	InvRef *ref = checkobject(L, 1);
+	Inventory *inv = getinv(L, ref);
+	if (!inv)
+		return 0;
+
+	const char *dst_listname = luaL_checkstring(L, 2);
+	const char *src_listname = luaL_checkstring(L, 3);
+
+	InventoryList *dst_list = inv->getList(dst_listname);
+	InventoryList *src_list = inv->getList(src_listname);
```

Comment:
`getList` may return `nullptr` is that list does not exist. Check for that.

Reply:
Will do.